internal/pkg/flink/internal/controller/application_controller.go

```diff
@@ -76,6 +81,10 @@ func (a *ApplicationController) ShowTableView() {
 	a.app.SetRoot(a.tableView, true).EnableMouse(false)
 }
 
+func (a *ApplicationController) AddCleanupFunction(cleanupFunction func()) {
```

Comment:
would be nice to make this a real builder pattern so that you get the same thing as the return e.g.

```suggestion
func (a *ApplicationController) AddCleanupFunction(cleanupFunction func()) *ApplicationController {
a.cleanupFunctions = append(a.cleanupFunctions, cleanupFunction)
return a
```

Reply:
Done! Needed to move the controller interface definitions to the types folder though because I got an import cycle in the mock app controller